src/plugins/dashboard/public/application/hooks/use_dashboard_app_state.ts

```diff
@@ -149,10 +151,14 @@ export const useDashboardAppState = ({
        */
       const dashboardSessionStorageState = dashboardSessionStorage.getState(savedDashboardId) || {};
       const dashboardURLState = loadDashboardUrlState(dashboardBuildContext);
+      const forwardedAppState: undefined | ForwardedDashboardState = scopedHistory()?.location
+        ?.state as undefined | ForwardedDashboardState;
+
       const initialDashboardState = {
         ...savedDashboardState,
         ...dashboardSessionStorageState,
         ...dashboardURLState,
+        ...forwardedAppState,
```

Comment:
This is where we add the state from `scopedHistory.location.state` to the dashboard state.